api/src/main/java/org/apache/iceberg/expressions/ExpressionUtil.java

```diff
@@ -291,7 +291,9 @@ private static String sanitize(Literal<?> literal) {
   }
 
   private static String sanitizeNumber(Number value, String type) {
-    int numDigits = (int) Math.log10(value.doubleValue()) + 1;
+    // log10 of zero isn't defined and will result in negative infinity
+    int numDigits =
+        0.0d == value.doubleValue() ? 1 : (int) Math.log10(Math.abs(value.doubleValue())) + 1;
```

Comment:
for `Double.MIN_VALUE` we'll get something like `(-322-digit-float)` because `Math.log10` produces a negative result. Not sure how much we care about `Double.MIN_VALUE`/`Float.MIN_VALUE` but we're not producing correct results for them

Reply:
Good point. Right now, this is summarizing the numeric value, even if that's outside the range of actual digits we'd be able to store because of precision. I think it's probably fine for now though?